src/sr_single_entity_handler.erl

```diff
@@ -0,0 +1,121 @@
+%%% @doc Base GET|PUT|DELETE /[entity]s/:id implementation
+-module(sr_single_entity_handler).
+
+-include_lib("mixer/include/mixer.hrl").
+-mixin([{ sr_entities_handler
+        , [ init/3
+          , allowed_methods/2
+          , content_types_provided/2
+          , announce_req/2
+          , handle_exception/3
+          ]
+        }]).
+
+-export([ rest_init/2
+        , resource_exists/2
+        , content_types_accepted/2
+        , handle_get/2
+        , handle_put/2
+        , delete_resource/2
+        ]).
+
+-type options() :: #{ path => string()
+                    , model => module()
+                    , verbose => boolean()
+                    }.
+-type state() :: #{ opts => options()
+                  , id => binary()
+                  , entity => sumo:user_doc()
+                  }.
+-export_type([state/0, options/0]).
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%% Cowboy Callbacks
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+-spec rest_init(cowboy_req:req(), options()) ->
+  {ok, cowboy_req:req(), state()}.
+rest_init(Req, Opts) ->
+  Req1 = announce_req(Req, Opts),
+  {Id, Req2} = cowboy_req:binding(id, Req1),
+  {ok, Req2, #{opts => Opts, id => Id}}.
+
+-spec resource_exists(cowboy_req:req(), state()) ->
+  {boolean(), cowboy_req:req(), state()}.
+resource_exists(Req, State) ->
+  #{opts := #{model := Model}, id := Id} = State,
+  case sumo:find(Model, Id) of
+    notfound -> {false, Req, State};
+    Entity -> {true, Req, State#{entity => Entity}}
+  end.
+
+%% @todo Use swagger's 'consumes' to auto-generate this if possible
+%% @see https://github.com/inaka/sumo_rest/issues/7
+-spec content_types_accepted(cowboy_req:req(), state()) ->
+  {[{{binary(), binary(), '*'}, atom()}], cowboy_req:req(), state()}.
+content_types_accepted(Req, State) ->
+  {[{{<<"application">>, <<"json">>, '*'}, handle_put}], Req, State}.
+
+-spec handle_get(cowboy_req:req(), state()) ->
+  {iodata(), cowboy_req:req(), state()}.
+handle_get(Req, State) ->
+  #{opts := #{model := Model}, entity := Entity} = State,
```

Comment:
Our experience dictates that it's impossible to get this function to throw an exception.